src/cls/rgw/cls_rgw.cc

```diff
@@ -3041,7 +3041,7 @@ static int usage_log_trim_cb(cls_method_context_t hctx, const string& key, rgw_u
   string key_by_time;
   string key_by_user;
 
-  string o = entry.owner.to_str();
+  string o = (entry.payer.empty() ? entry.owner.to_str() : entry.payer.to_str());
```

Comment:
this looks right, can you add a test case also, see https://github.com/ceph/ceph/pull/19322 and https://github.com/ceph/ceph/pull/19131 for some examples